core/symbol/match_spec.rb

```diff
@@ -24,3 +24,20 @@
     it_behaves_like :symbol_match, :match
   end
 end
+
+ruby_version_is "2.4" do
+  describe "Symbol#match" do
+    it "returns the MatchData" do
+      :abc.match(/b/)[0].should == 'b'
```

Comment:
I think it would be useful to test if `is_a?(MatchData)` here, like
```ruby
result = :abc.match(/b/)
result.should be_kind_of(MatchData)
result[0].should == 'b' 
```